src/meta/meta_bulk_load_ingestion_context.cpp

```diff
@@ -0,0 +1,195 @@
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "meta_bulk_load_ingestion_context.h"
+
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/utility/fail_point.h>
+
+namespace dsn {
+namespace replication {
+
+DSN_DEFINE_uint32("meta_server",
+                  bulk_load_node_max_ingesting_count,
+                  4,
+                  "max partition_count executing ingestion for one node at the same time");
+DSN_TAG_VARIABLE(bulk_load_node_max_ingesting_count, FT_MUTABLE);
+
+DSN_DEFINE_uint32("meta_server", bulk_load_node_min_disk_count, 1, "min disk count of one node");
+DSN_TAG_VARIABLE(bulk_load_node_min_disk_count, FT_MUTABLE);
+
+ingestion_context::ingestion_context() { reset_all(); }
+
+ingestion_context::~ingestion_context() { reset_all(); }
+
+void ingestion_context::partition_node_info::create(const partition_configuration &config,
+                                                    const config_context &cc)
+{
+    pid = config.pid;
+    std::unordered_set<rpc_address> current_nodes;
+    current_nodes.insert(config.primary);
+    for (const auto &secondary : config.secondaries) {
+        current_nodes.insert(secondary);
+    }
+    for (const auto &node : current_nodes) {
+        std::string disk_tag;
+        if (cc.get_disk_tag(node, disk_tag)) {
+            node_disk[node] = disk_tag;
+        }
+    }
+}
+
+void ingestion_context::node_context::init_disk(const std::string &disk_tag)
+{
+    if (disk_ingesting_counts.find(disk_tag) != disk_ingesting_counts.end()) {
+        return;
+    }
+    disk_ingesting_counts[disk_tag] = 0;
+}
+
+uint32_t ingestion_context::node_context::get_max_disk_ingestion_count(
+    const uint32_t max_node_ingestion_count) const
+{
+    FAIL_POINT_INJECT_F("ingestion_node_context_disk_count", [](string_view count_str) -> uint32_t {
+        auto count = 0;
+        buf2uint32(count_str, count);
+        return count;
+    });
+
+    const auto node_disk_count = disk_ingesting_counts.size() > FLAGS_bulk_load_node_min_disk_count
+                                     ? disk_ingesting_counts.size()
+                                     : FLAGS_bulk_load_node_min_disk_count;
+    return (max_node_ingestion_count + node_disk_count - 1) / node_disk_count;
+}
+
+bool ingestion_context::node_context::check_if_add(const std::string &disk_tag)
```

Comment:
```suggestion
bool ingestion_context::node_context::check_if_add(const std::string &disk_tag) const
```

Reply:
This function use disk_ingesting_counts[disk_tag] simplifier, it can't be const.